woocommerce/woocommerce-gateway-stripe
# UPE: allow toggling UPE on/off and toggling which UPE payment methods are enabled

# Changes proposed in this Pull Request:

These changes add a feature flag for UPE, allow turning UPE on and off, hiding and showing the LPM gateways depending if UPE is enabled, and turning the individual UPE methods on and off.

Additionally, when UPE is enabled, any legacy payment methods that were enabled automatically have their UPE equivalent enabled. Likewise when UPE is disabled, any UPE methods that were enabled automatically have their legacy equivalents methods enabled. This way, enabling or disabling UPE should never actually change which payment methods are available to customers.

# Testing instructions

- Enable UPE feature flag:
```
`INSERT INTO `wp_options` (`option_id`, `option_name`, `option_value`, `autoload`) VALUES (NULL, '_wcstripe_feature_upe', '1', 'no') ON DUPLICATE KEY UPDATE `option_value`=VALUES(`option_value`);`
```
- Go to WooCommerce > Settings > Payments
- Enable Stripe, EPS, Giropay
- Go into Stripe settings
- Check the "Enable new checkout experience" option and hit save.
- Scroll to the bottom and there should now be a table of UPE payment methods you can turn on and off. Card, EPS, and Giropay should be enabled because those LPMs were enabled when you turned UPE on.
- Go back to the main Settings > Payments screen. The methods that are managed by UPE should not be shown anymore.
- Uncheck the "Enable new checkout experience" option, save, and it should revert just fine, re-enabling the proper LPMs.
- Try other combinations of enabling and disabling LPMs, UPE, and UPE payment methods.
